roles/shared_handlers/handlers/main.yml

```
@@ -1,5 +1,10 @@
 # This role holds reusable handlers that can be included in multiple playbooks to keep things DRY
 
+- name: check service status
+  service_facts:
+  listen:
+    - restart sidekiq
+
```

I am not what this is used for ? Isn't that check done by this line below
```
 when: ansible_facts.services['sidekiq.service']['state'] == 'running'
```
I am not really familiar with ansible so I might be missing context.

---

It took me a while to understand it too, probably should add comments.

```suggestion
- name: fetch service status # into the global variable ansible_facts.services
  service_facts:
  listen: # run if the following tasks have been notified:
    - restart sidekiq

```

---

Yeah that makes it easier to understand :+1: